app/services/course/assessment/submission/csv_download_service.rb

```diff
@@ -0,0 +1,130 @@
+# frozen_string_literal: true
+require 'csv'
+class Course::Assessment::Submission::CsvDownloadService
+  class << self
+    # Downloads the submissions in csv format
+    #
+    # @param [CourseUser] current_course_user The course user downloading the submissions.
+    # @param [Course::Assessment] assessment The assessments to download submissions from.
+    # @param [String|nil] course_users_type The subset of course users whose submissions to download.
+    # Accepted values: 'my_students', 'my_students_w_phantom', 'students', 'students_w_phantom'
+    #   'staff', 'staff_w_phantom'
+    # @return [String] The path to the csv file.
+    def download(current_course_user, assessment, course_users_type)
+      service = new(current_course_user, assessment, course_users_type)
```

Comment:
This is quite smart haha, using the class method to instantiate an instance so that instance methods can be used. Is there any reason why this is being done, vs just using all class methods?

Reply:
haha hmm no specific reason. I think either way works and it seems like both ways are used in our codebase